eth/ethconfig/config.go

```diff
@@ -64,7 +63,6 @@ var Defaults = Config{
 	TxLookupLimit:      2350000,
 	TransactionHistory: 2350000,
 	StateHistory:       params.FullImmutabilityThreshold,
-	StateScheme:        rawdb.HashScheme,
```

Comment:
Well, we _could_ leave it, if we do something like this

```
func ParseStateScheme(provided, default string, disk ethdb.Database) (string, error) {
```
and call it via 
```golang
provided = ""
if ctx.IsSet(StateSchemeFlag.Name){ 
  provided = ctx.String(StateSchemeFlag.Name) 
}
scheme, err := rawdb.ParseStateScheme(provided, ctx.String(StateSchemeFlag.Name), disk)
```

I don't know if it's worth it though

Reply:
FWIW, perhaps it's better to *not* have HashScheme set in configs unless teh user explicitly does so. Otherwise someone who doesn't care and just did a dumpconfig will forever be stuck on Hash since it will keep forcing it. Otherwise if they don't care and dumpconfig omits it, they would switch to path when the default flips too.

Reply:
I mean when they resync. I can imagine people keeping configs around longer than their chain.